src/tests/recommender.spec.ts

```diff
@@ -0,0 +1,614 @@
+		const mockUser1: User = {
+			id: 1,
+			email: 'john@bob.dk',
+			firstName: 'john',
+			lastName: 'bob',
+			password: 'admin',
+			roles: ['ADMIN'],
+			createdAt: new Date(),
+			updatedAt: new Date(),
+			tutorInfo: null,
+			phoneNumber: '11223344',
+			education: '',
+			address: '',
+			dateOfBirth: new Date(),
+			avatarUrl: '',
+			languages: ['', ''],
+			subjectsOfInterest: ['', ''],
+			ratings: [new Rating(), new Rating()],
+			recommendations: [new Recommendation(), new Recommendation()],
+		};
+
+		const mockUser2: User = {
+			id: 2,
+			email: 'john@bob.dk',
+			firstName: 'john',
+			lastName: 'bob',
+			password: 'admin',
+			roles: ['ADMIN'],
+			createdAt: new Date(),
+			updatedAt: new Date(),
+			tutorInfo: null,
+			phoneNumber: '11223344',
+			education: '',
+			address: '',
+			dateOfBirth: new Date(),
+			avatarUrl: '',
+			languages: ['', ''],
+			subjectsOfInterest: ['', ''],
+			ratings: [new Rating(), new Rating()],
+			recommendations: [new Recommendation(), new Recommendation()],
+		};
+
+		const mockUser3: User = {
+			id: 3,
+			email: 'john@bob.dk',
+			firstName: 'john',
+			lastName: 'bob',
+			password: 'admin',
+			roles: ['ADMIN'],
+			createdAt: new Date(),
+			updatedAt: new Date(),
+			tutorInfo: null,
+			phoneNumber: '11223344',
+			education: '',
+			address: '',
+			dateOfBirth: new Date(),
+			avatarUrl: '',
+			languages: ['', ''],
+			subjectsOfInterest: ['', ''],
+			ratings: [new Rating(), new Rating()],
+			recommendations: [new Recommendation(), new Recommendation()],
+		};
+
+		const service1: Service = {
+			id: 1,
+			name: 'service1',
+			description: 'I am a service',
+			categories: null,
+			tutorInfo: null,
+			ratings: [new Rating(), new Rating()],
+			recommendations: [new Recommendation(), new Recommendation()],
+			image: '',
+		};
+		const service2: Service = {
+			id: 2,
+			name: 'service1',
+			description: 'I am a service',
+			categories: null,
+			tutorInfo: null,
+			ratings: [new Rating(), new Rating()],
+			recommendations: [new Recommendation(), new Recommendation()],
+			image: '',
+		};
+		const service3: Service = {
+			id: 3,
+			name: 'service1',
+			description: 'I am a service',
+			categories: null,
+			tutorInfo: null,
+			ratings: [new Rating(), new Rating()],
+			recommendations: [new Recommendation(), new Recommendation()],
+			image: '',
+		};
+
+		const allUsers: User[] = [];
+		allUsers.push(mockUser1);
+		allUsers.push(mockUser2);
+		allUsers.push(mockUser3);
+
+		const allServices: Service[] = [];
+		allServices.push(service1);
+		allServices.push(service2);
+		allServices.push(service3);
+
+		const userServiceMatrix: number[][] = [];
+		userServiceMatrix.push([null, 1, 2, 3]);
+		userServiceMatrix.push([1, 999, 999, 999]);
+		userServiceMatrix.push([2, 999, 999, 999]);
+		userServiceMatrix.push([3, 999, 999, 999]);
+
+		const res = initUserServiceMatrix(3, 3, allUsers, allServices, userServiceMatrix);
+
+		expect(res[1][1]).to.not.equal(999);
```

Comment:
Giver det mere mening at teste om de er lig 0 i stedet da det er det funktionen gør? Det har du også gjort i testen nedenunder🤔

Reply:
Det er bare for at teste at de faktisk bliver ændret, og 999 tænkte jeg aldrig ville forekomme, så den skiller sig ud